lib/widgets/workspace.dart

```diff
@@ -242,89 +378,109 @@ class _FilesWorkspaceState extends State<FilesWorkspace> {
     return baseString;
   }
 
+  void _onEntityTap(EntityInfo entity) {
+    final bool selected = controller.selectedItems.contains(entity);
+    final Set<LogicalKeyboardKey> keysPressed =
+        RawKeyboard.instance.keysPressed;
+    final bool multiSelect = keysPressed.contains(
+          LogicalKeyboardKey.controlLeft,
+        ) ||
+        keysPressed.contains(
+          LogicalKeyboardKey.controlRight,
+        );
+
+    if (!multiSelect) controller.clearSelectedItems();
+
+    if (!selected && multiSelect) {
+      controller.addSelectedItem(entity);
+    } else if (selected && multiSelect) {
+      controller.removeSelectedItem(entity);
+    } else {
+      controller.addSelectedItem(entity);
+    }
+    setState(() {});
+  }
+
+  void _onEntityDoubleTap(EntityInfo entity) {
+    if (entity.isDirectory) {
+      controller.currentDir = entity.path;
+    } else {
+      launch(entity.path);
+    }
+  }
+
+  // For move more than one file
+  void _onDropAccepted(String path) {
+    for (final entity in controller.selectedItems) {
+      Utils.moveFileToDest(entity.entity, path);
+    }
+  }
+
   Widget get body {
     return Builder(
       builder: (context) {
         if (controller.currentInfo != null) {
           if (controller.currentInfo!.isNotEmpty) {
-            return FilesTable(
-              rows: controller.currentInfo!
-                  .map(
-                    (entity) => FilesRow(
-                      entity: entity,
-                      selected: controller.selectedItems.contains(entity),
-                      onTap: () {
-                        bool selected =
-                            controller.selectedItems.contains(entity);
-                        final Set<LogicalKeyboardKey> keysPressed =
-                            RawKeyboard.instance.keysPressed;
-                        final bool multiSelect = keysPressed.contains(
-                              LogicalKeyboardKey.controlLeft,
-                            ) ||
-                            keysPressed.contains(
-                              LogicalKeyboardKey.controlRight,
-                            );
-
-                        if (!multiSelect) controller.clearSelectedItems();
-
-                        if (!selected && multiSelect) {
-                          controller.addSelectedItem(entity);
-                        } else if (selected && multiSelect) {
-                          controller.removeSelectedItem(entity);
-                        } else {
-                          controller.addSelectedItem(entity);
-                        }
-                        setState(() {});
-                      },
-                      onDoubleTap: () {
-                        if (entity.isDirectory) {
-                          controller.currentDir = entity.path;
-                        } else {
-                          launch(entity.path);
-                        }
-                      },
+            switch (controller.view) {
```

Comment:
continuing from the grid.dart file comment, the proposed approach would also benefit performance as exchanging widget is more expensive than modifying an existing one (related to Element tree)

Reply:
Good, but that logic will won't disappear, because otherwise we will not be able to change the workspace view. Or I just didn’t see other way to create this

Reply:
yea no the logic is fine, the important thing is that you switch what you pass to the widget and not the widget itself